gettsim/tests/test_rounding.py

```diff
@@ -0,0 +1,187 @@
+from gettsim.policy_environment import load_reforms_for_date
+from gettsim.shared import add_rounding_spec
+
+
+def test_decorator():
+    @add_rounding_spec(params_key="params_key_test")
+    def test_func():
+        return 0
+
+    assert test_func.__rounding_params_key__ == "params_key_test"
+
+
+@pytest.mark.parametrize(
+    "rounding_specs",
+    [
+        {},
+        {"params_key_test": {}},
+        {"params_key_test": {"rounding": {}}},
+        {"params_key_test": {"rounding": {"test_func": {}}}},
+    ],
+)
+def test_fail_if_no_rounding_specs(rounding_specs):
+    """Missing rounding specifications"""
+
+    with pytest.raises(KeyError):
+
+        @add_rounding_spec(params_key="params_key_test")
+        def test_func():
+            return 0
+
+        compute_taxes_and_transfers(
+            data=pd.DataFrame([{"p_id": 1}, {"p_id": 2}]),
+            functions=[test_func],
+            params=rounding_specs,
+            targets=["test_func"],
+        )
+
+
+@pytest.mark.parametrize(
+    "base, direction", [(1, "upper"), ("0.1", "down"), (5, "closest")],
+)
+def test_fail_if_rounding_specs_wrong_format(base, direction):
+    """Unsupported rounding specifications"""
+    with pytest.raises(ValueError):
+
+        @add_rounding_spec(params_key="params_key_test")
+        def test_func():
+            return 0
+
+        rounding_specs = {
+            "params_key_test": {
+                "rounding": {"test_func": {"base": base, "direction": direction}}
+            }
+        }
+
+        compute_taxes_and_transfers(
+            data=pd.DataFrame([{"p_id": 1}, {"p_id": 2}]),
+            functions=[test_func],
+            params=rounding_specs,
+            targets=["test_func"],
+        )
+
+
+@pytest.mark.parametrize(
+    "base, direction, input_values, exp_output",
+    [
+        (1, "up", [100.24, 100.78], [101.0, 101.0]),
+        (1, "down", [100.24, 100.78], [100.0, 100.0]),
+        (1, "nearest", [100.24, 100.78], [100.0, 101.0]),
+        (5, "up", [100.24, 100.78], [105.0, 105.0]),
+        (0.1, "down", [100.24, 100.78], [100.2, 100.7]),
+        (0.001, "nearest", [100.24, 100.78], [100.24, 100.78]),
+    ],
+)
+def test_rounding(base, direction, input_values, exp_output):
+    """Check if rounding is correct"""
+
+    # Define function that should be rounded
+    @add_rounding_spec(params_key="params_key_test")
+    def test_func(income):
+        return income
+
+    data = pd.DataFrame([{"p_id": 1}, {"p_id": 2}])
+    data["income"] = input_values
+    rounding_specs = {
+        "params_key_test": {
+            "rounding": {"test_func": {"base": base, "direction": direction}}
+        }
+    }
+
+    calc_result = compute_taxes_and_transfers(
+        data=data, functions=[test_func], params=rounding_specs, targets=["test_func"],
+    )
+    np.array_equal(calc_result["test_func"].values, np.array(exp_output))
+
+
+@pytest.mark.parametrize(
+    "base, direction, input_values, exp_output",
```

Comment:
```suggestion
    "base, direction, input_values_exp_output, _ignore",
```

Then re-use the parametrization from `test_rounding`.

Reply:
What would be the function of `_ignore`? If I understand your comment correctly, I could re-use the parametrization without an additional keyword.

Reply:
Ah, got it now :)